ph5/utilities/utmtolatlong.py

```diff
@@ -0,0 +1,167 @@
+                if hemisphere != "S" and hemisphere != "N":
+                    print ("Error, zone must be either N or S for \
+northern/southern hemisphere, but you used ", hemisphere)
+                    return
+
+                utmi = utm_conversions(easting, northing, zone, hemisphere)
+                lon, lat = utmi.utm_to_latlong()
+                print ("easting=%.1f, northing=%.1f, zone=%d, hemisphere=%s"
+                       % (easting, northing, zone, hemisphere))
+                print ("Lon = %.7f, Lat = %.7f" % (lon, lat))
+                outstr = "%.7f, %.7f" % (lon, lat) + "\n"
+                output.write(outstr)
+
+        finally:
+            dater.close()
+            output.close()
+
+    def doinline(self, args):
+
+        easting = float(args.easting)
+        northing = float(args.northing)
+        if easting < 0.0 or northing < 0.0:
+            print ("Error, easting and northing must be >0, but you used ",
+                   easting, " and ", northing)
+            return
+        try:
+            zone = int(args.zone)
+        except (RuntimeError, TypeError, NameError, ValueError):
+            print ("Error, zone must be an integer from 1-60, but you used ",
+                   args.zone)
+            return
+        if zone < 1 or zone > 60:
+            print ("Error, zone must be from 1 - 60, but you used ", zone)
+            return
+        hemisphere = str(args.side).upper()
+        if hemisphere != "S" and hemisphere != "N":
+            print ("Error, zone must be either N or S for northern/southern \
+hemisphere, but you used ", hemisphere)
+            return
+
+        print ("easting=%.1f, northing=%.1f, zone=%d, hemisphere=%s"
+               % (easting, northing, zone, hemisphere))
+        utmi = utm_conversions(easting, northing, zone, hemisphere)
+        lon, lat = utmi.utm_to_latlong()
+        result = "Lon = %.7f, Lat = %.7f" % (lon, lat)
+        print (result)
+        print ("***************************************")
+
+
+def main():
+
+    desc = 'Convert UTM locations to longitudes, latitudes, \
+in batches (-i, -o) or inline (-e,-n,-z,-s)'
+    parser = argparse.ArgumentParser(description=desc)
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-i", "--infile", help="Input file with UTM \
+Easting, Northing, Zone, Hemisphere")
+    parser.add_argument("-o", "--outfile", help="Output file with \
+converted longitudes, latitudes")
+    group.add_argument("-e", "--easting", help="Enter UTM Easting")
+    parser.add_argument("-n", "--northing", help="Enter UTM Northing")
+    parser.add_argument("-z", "--zone", help="Enter UTM Zone")
+    parser.add_argument("-s", "--side", help="Side=Hemisphere, N or S")
+
+    args = parser.parse_args()
+    # print ("args:", args)
+    utmi = utm_conversions()
+
+    try:
+        fli = len(args.infile)
+        flo = len(args.outfile)
+    except (RuntimeError, TypeError, NameError, ValueError):

```

Comment:
Why is the try/except necessary?

Reply:
I will get the class names nd comments cleaned up, and will set encoding to UTF-8. I'm using Gedit for Mac, I didn't see the ^M characters.
The try/except is necessary if the user uses the inline `-e Easting -n Northing etc.` method, which is distinct from the batch file method. In the inline case, the variables for infile and outfile are never defined, and testing them for length raises an error. They are only defined if user uses the `-i ifile -o outfile` option. Thus the try/catch.
Going to a funeral, will do another push later this morning.

Reply:
> In the inline case, the variables for infile and outfile are never defined, and testing them for length raises an error. They are only defined if user uses the -i ifile -o outfile option.

Instead of catching these errors you should explicitly check if `args.infile` and `args.outfile` exist. If they exist then take their length, else set `fli` and `flo` to zero.

Reply:
> I'm using Gedit for Mac, I didn't see the ^M characters.

You can see these carriage returns in GitHub (https://github.com/PIC-IRIS/PH5/pull/321/files).